src/ort_genai_c.h

```diff
@@ -151,15 +154,100 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaGenerate(const OgaModel* model, const OgaG
  */
 OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateGeneratorParams(const OgaModel* model, OgaGeneratorParams** out);
 
+/*
+ * \brief Creates Lora Adapter within a given model that one can add
+ *        parameters to. The adapter_name must be unique within the model.
+ * \param[in] model The model to create adapter with.
+ * \param[in] adapter_name to be created
+ * \return OgaResult containing the error message if the adapter creation failed.
+ */
+OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateLoraAdapter(OgaModel* model, const char* adapter_name);
```

Comment:
I'm thinking if we need a LoraAdapterManager. There will be 2 concepts/classes: LoraAdaperManager and LoraAdapter.
LoraAdapterManager mamanges the LoraAdapter, which has functionalities like below:
- AddLoraAdapter and AddLoraAdapters
- RemoveLoraAdapter and RemoveLoraAdapters
- GetLoraAdapter
And it also provides cache logic

LoraAdapter contains the all the data for one Adapter, mainly adapter_name & Tensor pair.

For the integration with GenAI API:

- Integration with Model. if we save adapters data together with the model folder, we can create the LoraAdapterManager from the model folder by loading the adapters data from the ModelFolder. if we decide to make adapters store in a separate folder, we can create the LoraAdapterManager from the seperate folder.
- integration with Generator. We need one API: SetLoraAdapter with LoraAdapter, which is retried from the LoraAdapterManager, as the parameter.

Reply:
There are two classes.
However, that way we would have to expose and extra 2 classes via public API and all the language bindings. On top of that, in C# LoraAdapterManager would have to be IDisposable which would put an extra burden on the client code.
The way I exposed it is via an existing Model class which does not require two more class interfaces.

Reply:
Do we envision sharing the adapters between multiple models? If yes, it makes sense to have a separate LoraAdapterManager class exposed as it's lifecycle will be different from the models using them.